packages/edit-site/src/components/sidebar/global-styles-v2/components/url-sync.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * WordPress dependencies
+ */
+import { getQueryArgs } from '@wordpress/url';
+import { useEffect, useRef } from '@wordpress/element';
+import { useNavigatorLocation } from '@wordpress/components';
+
+const PARAM = 'gssb';
+
+export const useInitialPath = () => {
+	const pathRef = useRef();
+	try {
+		const parsed = getQueryArgs( window.location.href );
```

Comment:
We're using `getQueryArgs()` from `@wordpress/url` to avoid having to bundle `query-string`.

Reply:
Why are we using the URL at call, can't we just keep the "path" as internal state?

Reply:
This is a good question! Most of the code in this PR comes directly from https://github.com/ItsJonQ/g2/pull/293 , and so we have little insight into the rationale behind the code, especially the one related to Navigation/Routing (in fact we had initially decided to keep this part of the prototype untouched, and come to it a later point).

Having said that, your question makes sense! I can spot a `MemoryRouter` component in `packages/components/src/navigator/router/memory-router.js`, that looks like something we may leverage in case we didn't want to modify the URL?

Reply:
I think our components in `@wordpress/components` shouldn't be exposing anything related to URLs at all as this doesn't make sense in a react native context. If we want to sync the navigation state to URLs, we can still do it in the application layer (edit-site) like done here but I don't see it as a requirement for a start.

Reply:
Let's move the conversation to #34907